### PR TITLE
feat: core.331 support.gcs.artifact.repository

### PR DESCRIPTION
**Pull request by aleksandrmelnikov**
**What this PR does**:
Added code to read GCS from artifactRepository, and properly set the placeholder values.
**Which issue(s) this PR fixes**:331

Fixes onepanelio/core#331

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/build.go [hunk 1]
```diff
@@ -173,6 +174,50 @@ func GenerateKustomizeResult(config opConfig.Config, kustomizeTemplate template.
...
+	artifactRepoS3Node, _ := yamlFile.Get("artifactRepository.s3")
+	if artifactRepoS3Node != nil {
+		_, keyFormat := yamlFile.Get("artifactRepository.s3.keyFormat")
+		_, bucket := yamlFile.Get("artifactRepository.s3.bucket")
+		_, endpoint := yamlFile.Get("artifactRepository.s3.endpoint")
+		_, insecure := yamlFile.Get("artifactRepository.s3.insecure")
+		_, region := yamlFile.Get("artifactRepository.s3.region")
+		_, accessKey := yamlFile.Get("artifactRepository.s3.accessKey")
+		_, secretKey := yamlFile.Get("artifactRepository.s3.secretKey")
+		artifactRepoS3NodeVal := fmt.Sprintf(""+
```
Comment (rushtehrani) [inline cmd/build.go:186]: Is there a way to do this by mutating the YAML tree instead of a big string like this?

Comment (aleksandrmelnikov) [reply cmd/build.go]: Changes are implemented.

File: cmd/build.go [hunk 2]
```diff
@@ -173,6 +175,45 @@ func GenerateKustomizeResult(config opConfig.Config, kustomizeTemplate template.
 		yamlFile.PutWithSeparator("metalLbSecretKey", base64.StdEncoding.EncodeToString(metalLbSecretKey), ".")
 	}
 
+	artifactRepoS3Node, _ := yamlFile.Get("artifactRepository.s3")
```
Comment (rushtehrani) [inline cmd/build.go:178]: Haven't fully tested this, but can this section be refactored like so?
```
        _, artifactRepositoryNode := yamlFile.Get("artifactRepository")
	artifactRepositoryConfig := v1.ArtifactRepositoryConfig{}
	artifactRepositoryNode.Decode(artifactRepositoryConfig)
	if err != nil {
		return "", err
	}
	if artifactRepositoryConfig.S3 != nil {
		artifactRepositoryConfig.S3.AccessKeySecret.Key = artifactRepositoryConfig.S3.AccessKey
		artifactRepositoryConfig.S3.AccessKeySecret.Name = "$(artifactRepositoryS3AccessKeySecretName)"
		artifactRepositoryConfig.S3.SecretKeySecret.Key = artifactRepositoryConfig.S3.Secretkey
		artifactRepositoryConfig.S3.SecretKeySecret.Name = "$(artifactRepositoryS3SecretKeySecretName)"
		err, yamlStr := artifactRepositoryConfig.S3.MarshalToYaml()
		if err != nil {
			return "", err
		}
		yamlFile.Put("artifactRepositoryProvider", yamlStr)
	}
	else if artifactRepositoryConfig.GCS != nil {
		err, yamlConfigMap := artifactRepositoryConfig.GCS.MarshalToYaml()
		if err != nil {
			return "", err
		}

		yamlFile.Put("artifactRepositoryProvider", yamlConfigMap)
	} else {
		return "", errors.New("unsupported artifactRepository configuration")
	}
```

Comment (rushtehrani) [reply cmd/build.go:178]: Also, it looks like GCS is still not using secrets?

Comment (aleksandrmelnikov) [reply cmd/build.go]: This is implemented.

File: cmd/build.go [hunk 3]
```diff
@@ -193,28 +221,30 @@ func GenerateKustomizeResult(config opConfig.Config, kustomizeTemplate template.
...
-		var stringToWrite = fmt.Sprintf("%v=%v\n%v=%v\n%v=%v\n%v=%v\n",
-			"artifactRepositoryBucket", flatMap["artifactRepositoryS3Bucket"],
-			"artifactRepositoryEndpoint", flatMap["artifactRepositoryS3Endpoint"],
-			"artifactRepositoryInsecure", flatMap["artifactRepositoryIS3nsecure"],
-			"artifactRepositoryRegion", flatMap["artifactRepositoryS3Region"],
-		)
-		_, err = paramsFile.WriteString(stringToWrite)
-		if err != nil {
-			return "", err
+	if yamlFile.HasKey("artifactRepository.s3") {
```
Comment (rushtehrani) [inline cmd/build.go:224]: This is a bit confusing, I see a check here for s3, but nothing for `gcs`. Maybe explain why we only check for `s3` here?

Comment (rushtehrani) [reply cmd/build.go:224]: Also, should we just check `artifactRepositoryConfig.S3 != nil` here too?

Comment (aleksandrmelnikov) [reply cmd/build.go]: Agreed, updated.

**Pull request merged**